hail/src/main/scala/is/hail/expr/ir/agg/CallStatsAggregator.scala

```diff
@@ -16,7 +18,7 @@ object CallStatsState {
   val callStatsInternalArrayType = PCanonicalArray(PInt32Required, required = true)
   val stateType: PCanonicalTuple = PCanonicalTuple(true, callStatsInternalArrayType, callStatsInternalArrayType)
 
-  val resultType = PCanonicalStruct(required = true,
+  val resultPType = PCanonicalStruct(required = false,
```

Comment:
Is this the right thing to do? I'm doing it because I can't construct a `SBaseStructPointer` of a `PCanonicalStruct` that is `required` (we have an assertion forbidding it). This is causing a small test failure in a requiredness test of nested aggregators where we have hardcoded that we expect it to be required. I still don't understand why we have these assertions.

Reply:
@tpoterba , Patrick said he is good with this PR but doesn't know the answer to this.

Reply:
this is the right thing to do, yes, if you're still using a PCanonicalStruct. I think we should follow up this change (which is fine as-is0 with an audit to ensure the results of the aggregators aren't doing undue allocation -- here we can certainly be using a stack struct!